packages/components/src/components/modal/modal.tsx

```diff
@@ -130,6 +130,8 @@ const ModalElement = ({
 
     const rendered_title = renderTitle ? renderTitle() : null;
 
+    const is_header_visible = !is_risk_warning_visible && (header || title || rendered_title || has_close_icon);
```

Comment:
please refer [this](https://github.com/binary-com/deriv-app/pull/8810/files/1a72bc59a40275f1f10ec40840e711347ba127d1#diff-205901201bf0e06d72552ba9926770c3102a740a96088321d17d4b1807feecc6) as original source
there you can find an explanation of current refactoring